eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/actions/AddToCalendarActionExecutable.java

```diff
@@ -72,7 +73,7 @@ private Bundle extrasForAddToCalendar()
         Bundle bundle = new Bundle(8);
         bundle.putLong(CalendarContract.EXTRA_EVENT_BEGIN_TIME, mEvent.start().getTimestamp());
         bundle.putLong(CalendarContract.EXTRA_EVENT_END_TIME,
-                mEvent.start().addDuration(mEvent.duration()).getTimestamp());
+                mEvent.start().addDuration(mEvent.duration().value(mEvent.start().isAllDay() ? new Duration(1, 1, 0) : Duration.parse("PT2H"))).getTimestamp());
```

Comment:
Maybe later. The default is really just needed when the event is added to the calendar. It shouldn't be necessary anywhere else in our own UI code. So this is really just a workaround. I hope that we'll be able to predict durations more accurately, eventually.